plugins/updater/src/updater.rs

```diff
@@ -239,6 +242,19 @@ impl UpdaterBuilder {
         self
     }
 
+    /// Allows you to modify the `reqwest`` client builder before the HTTP request is sent.
+    ///
+    /// Note that `reqwest` crate may be updated in minor releases of tauri-plugin-updater.
+    /// Therefore it's recommended to pin the plugin to at least a minor version when you're using `on_before_request`.
+    ///
+    pub fn on_before_request<F: Fn(ClientBuilder) -> ClientBuilder + Send + Sync + 'static>(
```

Comment:
I think this can be named `configure_client` and we should call it when downloading the update as well  not just when checking.